### PR TITLE
Add fluid.md

### PR DESCRIPTION
**Pull request by wangkuiyi**

### CODE REVIEW

File: doc/design/fluid.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+When Baidu developed PaddlePaddle in 2013, the only well-known open source deep learning system is Caffe.  However, when it open-sourced PaddlePaddle in 2016, there had been many other choices over there.  We were facing a challenge -- why would we open source yet another one?  
+
+Fluid is the answer.  Fluid is similar to PyTorch and TensorFlow Eager Execution, which describes the "process" of training or inference a model, but not the model itself.  Indeed, in PyTorch, Eager Execution, and Fluid, there is no such a concept of the model at all. I will explain in this article, Fluid is currently more extreme in this idea than PyTorch and Eager Execution, and we are pushing Fluid towards a compiler and even a new programming language for deep learning
+
+## The Evolution of Deep Learning Systems
+
+Deep learning infrastructure is one of the fastest involving technology.  Within only four years, there have been three generations of technologies invented. 
+
+| Since around | model = equence of layers | model = graph of operators | No model |
```
Comment (tonyyang-svail) [inline doc/design/fluid.md]: typo: equence => sequence

File: doc/design/fluid.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Design Doc: PaddlePaddle Fluid
+
+## Why Fluid
+
+When Baidu developed PaddlePaddle in 2013, the only well-known open source deep learning system is Caffe.  However, when it open-sourced PaddlePaddle in 2016, there had been many other choices over there.  We were facing a challenge -- why would we open source yet another one?
```
Comment (abhinavarora) [inline doc/design/fluid.md]: This should be, *the only well-known open source deep learning system **was** Caffe*

File: doc/design/fluid.md (hunk 1 above)
Comment (wangkuiyi) [reply doc/design/fluid.md]: done

File: doc/design/fluid.md (hunk 2 above)
Comment (wangkuiyi) [reply doc/design/fluid.md]: done

**Comment by kavyasrinet** [comment]
Merging this and creating a new PR.

**Pull request merged**